code/modules/mob/living/carbon/human/human_powers.dm

```diff
@@ -1336,3 +1336,837 @@
+				target.remove_language(LANGUAGE_VAURCA)
+				S.banned = TRUE
+				host.last_action = world.time + 1 MINUTES
+		else if(S.shielded == SOCKET_FULLSHIELDED)
+			to_chat(src, SPAN_WARNING("You feel a sudden pain in your neural socket as you are repelled by [target]'s countermeasures!"))
+			to_chat(target, SPAN_WARNING("[src] attempted to ban you from the Hivenet, but your countermeasures repelled them!"))
+			src.adjustHalLoss(15)
+			src.flash_pain(15)
+			host.last_action = world.time + 10 MINUTES
+			return
+
+/mob/living/carbon/human/proc/hivevoid() //Destroys a Vaurca's neural socket.
+	set name = "Void Hivenet User"
+	set desc = "Permanently sever a Vaurca of your own Hive from the Hivenet, destroying their neural socket. Mouv Ta may attempt to use this on Vaurca of another Hive, though this may have severe diplomatic consequences!"
+	set category = "Hivenet"
+	var/list/available_vaurca = list()
+	var/list/fullname = splittext(src.name, " ")
+	var/surname = fullname[2]
+	var/obj/item/organ/internal/vaurca/neuralsocket/host = src.internal_organs_by_name[BP_NEURAL_SOCKET]
+	if(src.stat != CONSCIOUS)
+		to_chat(src, SPAN_WARNING("You are incapable of that in your current state!"))
+		return
+	if(!(all_languages[LANGUAGE_VAURCA] in src.languages) || !istype(host))
+		to_chat(src, "<span class='danger'>Your mind is dark, unable to communicate with the Hive.</span>")
+		return
+	if(!istype(src.internal_organs_by_name[BP_NEURAL_SOCKET], /obj/item/organ/internal/vaurca/neuralsocket/admin))
+		to_chat(src, "<span class='warning'>You lack the authority to do this!.</span>")
+		return
+	for(var/mob/living/carbon/human/player in player_list)
+		if(isvaurca(player) && player.internal_organs_by_name[BP_NEURAL_SOCKET])
+			var/list/player_fullname = splittext(player.name, " ")
+			var/player_surname = player_fullname[2]
+			if(player_surname == surname || istype(src.origin, /singleton/origin_item/origin/mouv_b))
+				LAZYADD(available_vaurca, player)
+	LAZYREMOVE(available_vaurca, src) //can't ban/mute/void yourself
+	var/mob/living/carbon/human/target = input(src, "Select a Vaurca to void.", "Hivenet Void") as null|anything in available_vaurca
+	if(!target || !isvaurca(target))
+		return
+	var/choice = alert(src, "Are you sure you want to void [target]? This cannot be undone!", "Void User?", "Proceed", "Cancel")
+	if(choice == "Cancel")
+		return
+	msg_admin_attack("[key_name_admin(src)] attempted to void [key_name_admin(target)]'s neural socket! (<A HREF='?_src_=holder;adminplayerobservecoodjump=1;X=[src.x];Y=[src.y];Z=[src.z]'>JMP</a>)",ckey=key_name(src),ckey_target=key_name(target))
+	if(host.last_action > world.time)
+		to_chat(src, SPAN_WARNING("You must wait before attempting another Hivenet adminisrative action!"))
```

Comment:
this check is in the wrong place + you need to add a return: it needs to be at the very start of the proc, otherwise it's pointless to get all the way down here - it's sending out a log that you did a thing but then it's erroring out

you also typo'd administrative